### PR TITLE
Fix issue with `externalTrafficPolicy: Local` for single-stack services on dual-stack nodes

### PR DESCRIPTION
**Pull request by brandond**
#### Proposed Changes ####

Fix issue with local traffic policy for single-stack services on dual-stack nodes.

Just enable IP forwarding for all address families regardless of service address families.

Note: I confirmed that this works even on a node with ipv6 disabled via `net.ipv6.conf.all.disable_ipv6=1` sysctl.

#### Types of Changes ####

bugfix

#### Verification ####

https://github.com/k3s-io/k3s/issues/9949#issuecomment-2059645672

#### Linked Issues ####

* https://github.com/k3s-io/k3s/issues/9949

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: pkg/cloudprovider/servicelb.go [hunk 1]
```diff
@@ -443,13 +435,9 @@ func (k *k3s) newDaemonSet(svc *core.Service) (*apps.DaemonSet, error) {
 	}
 	sourceRanges := strings.Join(sourceRangesSet.StringSlice(), ",")
 
-	var sysctls []core.Sysctl
 	for _, ipFamily := range svc.Spec.IPFamilies {
 		switch ipFamily {
```
Comment (manuelbuil) [inline pkg/cloudprovider/servicelb.go:439]: We don't need a switch anymore, or?

Comment (brandond) [reply pkg/cloudprovider/servicelb.go:439]: I guess I could simplify to just an if ;)

**Review by manuelbuil — APPROVED** [review]

**Comment by damdo** [comment]
@brandond @manuelbuil is this ready to merge now? Thanks!

**Comment by manuelbuil** [comment]
> @brandond @manuelbuil is this ready to merge now? Thanks!

We are currently on code freeze. We can't merge anything

**Comment by damdo** [comment]
I see @manuelbuil thanks.
My k3s cluster is currently impacted by this issue.
I'd appreciate if this could merge when the code freeze lifts, and whether this could be backported to 1.29.z.
Thanks!

**Review by matttrach — APPROVED** [review]
lgtm

**Pull request merged**